services/transaction-statistics/shared/buildTransactionStatistics.js

```diff
@@ -0,0 +1,251 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const BluebirdPromise = require('bluebird');
+const moment = require('moment');
+const BigNumber = require('big-number');
+
+// Rename all methods once MySQL implementation is updated
+const {
+	Logger,
+	Queue,
+	MySQL: {
+		getDbConnection,
+		getTableInstance,
+		startDbTransaction,
+		commitDbTransaction,
+		rollbackDbTransaction,
+	},
+	Signals,
+} = require('lisk-service-framework');
+
+const { getDistributionByType } = require('./transactionStatistics');
+const { resolveGlobalTokenID, DB_CONSTANT, DATE_FORMAT } = require('./utils/constants');
+const { requestIndexer } = require('./utils/request');
+const requestAll = require('./utils/requestAll');
+
+const txStatisticsIndexSchema = require('./database/schemas/transactionStatistics');
+const config = require('../config');
+
+const logger = Logger();
+
+const MYSQL_ENDPOINT = config.endpoints.mysql;
+
+const getDBInstance = () => getTableInstance(
+	txStatisticsIndexSchema.tableName,
+	txStatisticsIndexSchema,
+	MYSQL_ENDPOINT,
+);
+
+const getTxStatsWithFallback = (acc, moduleCommand, range) => {
+	const defaultValue = {
+		count: 0,
+		volume: 0,
+		tokenID: DB_CONSTANT.UNAVAILABLE,
+	};
+	return acc[moduleCommand]
+		? acc[moduleCommand][range] || defaultValue
+		: defaultValue;
+};
+
+const getTxValue = tx => BigNumber(tx.params.amount).plus(tx.fee);
+
+const getRange = tx => {
+	const value = getTxValue(tx);
+	const lowerBoundExponent = Math.floor(Math.log10(value / 1e8));
```

Comment:
Can other tokens have different precision? If not, better to use a constant instead of hard-coded `1e8`.

Reply:
Added comment to handle this.